### DISCUSSION TITLE
Ignoring new major versions of base images when using dockerfile manager

### DISCUSSION THREAD

**Discussion by rvillane** [renovatebot/renovate#18683] [Request Help] [answered]
Hi, I'm using the dockerfile manager to keep a nodeJS builder container up to date with new minor or patch updates that in the context of dockerfile manager means newer sha256 versions.

FROM node:16-bullseye**@sha256:5f8e2399cc1866823d21ef9ebbdad9cdcf31dbf093f94b85600086ca408643d9**

The problem I'm having is that when new major version of node are released, renovate creates a PR suggesting to update to the latest version available, node 19 at the moment of this writing, but I'm not interested on those major versions.

Do you have recommendations about how to configure the renovate bot to only care about minor and patch versions when using the dockerfile manager ?

I think is relevant to mention that when using NPM manager, having PR's with new major versions of libraries is fine

thanks for your help

**Comment by rarkins** [comment, marked as answer]
You'll need a package rule with `allowedVersions`. Searching this repo for `node` and `allowedVersions` will reveal some past discussions: https://github.com/renovatebot/renovate/search?q=node+allowedversions&type=discussions

**Reply by rvillane** [reply]
Ok, I have made significant process with this `allowedVersions` setting, thanks for the tip.  Is there a way to have a general rule using allowedVersion to say, node version has to be lower than 19

```
  packageRules: [
    {
      "matchPackageNames": ["node"],
      "allowedVersions": "<19"
    },    
  ],
```

but in some specific repo, override that general rule and say "node" exactly equals to version "16" ?

**Reply by viceice** [reply]
yes, use shared preset's.

**Reply by rvillane** [reply]
the combination of both `allowedVersions` and shared presets provided the solution I was looking for, thank you very much
